py/server/deephaven/rollup_table.py

```diff
@@ -0,0 +1,195 @@
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_node_ops_recorder
+
+    def __init__(self, j_node_ops_recorder: jpy.JType):
+        self.j_node_ops_recorder = j_node_ops_recorder
+
+
+class RollupTable(JObjectWrapper):
+    """ A RollupTable is generated as a result of applying the :meth:`~deephaven.table.Table.rollup` operation on a
+    :class:`~deephaven.table.Table`.
+
+    Note: RollupTable should not be instantiated directly by user code.
+    """
+    j_object_type = _JRollupTable
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_rollup_table
+
+    def __init__(self, j_rollup_table: jpy.JType, aggs: Sequence[Aggregation], include_constituents: bool,
+                 by: Sequence[str]):
+        self.j_rollup_table = j_rollup_table
+        self.aggs = aggs
+        self.include_constituents = include_constituents
+        self.by = by
+
+    def node_operation_recorder(self, node_type: NodeType) -> RollupNodeOperationsRecorder:
+        """Creates a RollupNodeOperationsRecorder for per-node operations to apply during snapshotting of this
+        RollupTable. The recorded node operations will be applied only to the node of the provided NodeType. See
+        :class:`NodeType` for details.
+
+        Args:
+            node_type (NodeType): the type of node tables that the recorded operations will be applied to; if it is
+                :attr:`NodeType.CONSTITUENT`, the RollupTable must be created with include_constituents=True.
+
+        Returns:
+            a RollupNodeOperationsRecorder
+
+        Raises:
+            DHError
+        """
+        try:
+            return RollupNodeOperationsRecorder(j_node_ops_recorder=self.j_rollup_table.makeNodeOperationsRecorder(
+                node_type.value))
+        except Exception as e:
+            raise DHError(e, "failed to create a RollupNodeOperationsRecorder.") from e
+
+    def with_node_operations(self, node_ops_recorders: List[RollupNodeOperationsRecorder]) -> RollupTable:
+        """Returns a new RollupTable that will apply the recorded node operations to nodes when gathering
+        snapshots.
+        
+        Args:
+            node_ops_recorders (List[RollupNodeOperationsRecorder]): a list of RollupNodeOperationsRecorder containing
+                the node operations to be applied
+
+        Returns:
+            a new RollupTable
+
+        Raises:
+            DHError
+        """
+        try:
+            return RollupTable(
+                j_rollup_table=self.j_rollup_table.withNodeOperations(
+                    [op.j_node_ops_recorder for op in node_ops_recorders]),
+                include_constituents=self.include_constituents, aggs=self.aggs, by=self.by)
+        except Exception as e:
+            raise DHError(e, "with_node_operations on RollupTable failed.") from e
+
+    def with_filters(self, filters: Union[str, Filter, Sequence[str], Sequence[Filter]]) -> RollupTable:
```

Comment:
Is there a reason this isn't named `where` to be consistent with the rest of the API?